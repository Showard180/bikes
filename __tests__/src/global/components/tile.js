import 'react-native';
import React from 'react';
import Tile from '../../../../src/global/components/tile';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders', () => {
  const tree = renderer.create(
    <Tile />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders title and text when given in props', () => {
  const tree = renderer.create(
    <Tile title="Testing" text="Test"/>
  )
  expect(tree).toMatchSnapshot();
});

test('button when given', () => {
  const tree = renderer.create(
    <Tile button={{onPress: () => console.log('hello'), title: 'Hello'}}/>
  )
  expect(tree).toMatchSnapshot();
});
