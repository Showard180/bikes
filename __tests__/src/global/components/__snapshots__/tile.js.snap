// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button when given 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": "auto",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "padding": 10,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 30,
          "marginBottom": 10,
          "textDecorationLine": "underline",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "display": "flex",
          "marginLeft": 30,
        }
      }
    />
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Hello
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": "auto",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "padding": 10,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 30,
          "marginBottom": 10,
          "textDecorationLine": "underline",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "display": "flex",
          "marginLeft": 30,
        }
      }
    />
  </View>
</View>
`;

exports[`renders title and text when given in props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": "auto",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "padding": 10,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 30,
          "marginBottom": 10,
          "textDecorationLine": "underline",
        }
      }
    >
      Testing
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "display": "flex",
          "marginLeft": 30,
        }
      }
    >
      Test
    </Text>
  </View>
</View>
`;
